torch/cuda/memory.py

```diff
@@ -1177,11 +1155,8 @@ def snapshot(self):
             See :ref:`cuda-memory-management` for more details about GPU memory
             management.
         """
-        try:
-            ctx = MemPoolContext(self)
-            snapshot = torch.cuda.memory_snapshot()
-        finally:
-            del ctx
+        # TODO fix snapshot to return this pool only
```

Comment:
Isn't this done? You do thread the id all the way down to filter things?

Reply:
Done indeed, will remove TODO